### DISCUSSION TITLE
lightdm-deepin-greeter 会在进入 非 deeipn 桌面环境时 显示 deepin 桌面壁纸

### DISCUSSION THREAD

**Discussion by electricface** [linuxdeepin/developer-center#3106] [Features Request & Ideas | 特性请求 & 头脑风暴]
https://bbs.deepin.org/forum.php?mod=viewthread&tid=173727&extra=

> 引导DDE以外的桌面环境时能不能把显示dde桌面这一块去掉，或者对于kde、xfce等做出适配

@wangjia12 @zccrs @hualet @justforlxz

**Comment by justforlxz** [comment]
我当初提过，被夏斌和李鹤拒绝了，说不用改 😶

**Comment by hualet** [comment]
怎么适配？

**Comment by justforlxz** [comment]
> 怎么适配？

greeter是知道启动的session名称的，可以硬匹配一下deepin才开启渐变效果。

**Comment by quank123wip** [comment]
Watching

**Comment by wangjia12** [comment]
> 我当初提过，被夏斌和李鹤拒绝了，说不用改

这对我们桌面有什么影响？为什么拒绝了？

**Comment by justforlxz** [comment]
他俩说先不考虑其他桌面环境😂

Wangjia <notifications@github.com> 于2019年3月21日周四 下午3:46写道：

**Comment by Hi-Phile** [comment]
Seeing a problem on Arch Linux (RebornOS) where the lightdm-deepin-greeter is displaying twice before login.  The first login greeter appears, generic screen, without a login name and the password field, then type the password and a second lightdm-deeping-greeter appears with the icon graphic, correct background, username filed in, then type the password will login to the Deepin Desktop.  This happened with the latest deepin updates.

**Comment by justforlxz** [comment]
> Seeing a problem on Arch Linux (RebornOS) where the lightdm-deepin-greeter is displaying twice before login. The first login greeter appears, generic screen, without a login name and the password field, then type the password and a second lightdm-deeping-greeter appears with the icon graphic, correct background, username filed in, then type the password will login to the Deepin Desktop. This happened with the latest deepin updates.

This is because greeter is part of the dde desktop, and we do not distinguish the desktop environment that is about to log in, so we will use the login wallpaper from the deepin desktop environment.
